admin/listPurchases/listView.html

```diff
@@ -100,51 +100,67 @@
     });
 });
 
-const amountOfShownSubpages = 5;
 const entriesPerSubpage = 20;
+const amountOfShownSubpages = 5;
 let amountOfSubpages;
 
 function generateSubPageNavigation(){
-    if(orderSearchBar.value){
-        amountOfSubpages = Math.ceil(amountOfEntries/ entriesPerSubpage);
-    }else{
-        amountOfSubpages = Math.ceil(amountOfEntries / entriesPerSubpage);
-    }
+    amountOfSubpages = Math.ceil(amountOfEntries/ entriesPerSubpage);
 
-    if(currentPageIndex >= amountOfSubpages){
-        currentPageIndex = amountOfSubpages - 1;
-    } 
+    const extraNavNeeded = amountOfSubpages > amountOfShownSubpages;
 
     let startIndexToRender = 0;
     let subpageNavigation = [];
 
-    const extraNavNeeded = amountOfSubpages > amountOfShownSubpages;
+    if(extraNavNeeded){
+        startIndexToRender = ShiftedStartIndexToRender();
+        addBackwardsExtraNav(subpageNavigation);
+    }
+
+    addDefaultNav(subpageNavigation, startIndexToRender);
 
     if(extraNavNeeded){
-        subpageNavigation.push(`<a href="#" class="option subpage-selector subpage-selector-to-start">&lt&lt</a>`);
-        subpageNavigation.push(`<a href="#" class="option subpage-selector subpage-selector-backwards">&lt</a>`);
-
-        startIndexToRender = currentPageIndex;
-        if(startIndexToRender + Math.floor(amountOfShownSubpages/2) >= amountOfSubpages){
-            startIndexToRender = amountOfSubpages - amountOfShownSubpages;
-        }else if(startIndexToRender - Math.floor(amountOfShownSubpages/2) > 0){
-            startIndexToRender -= Math.floor(amountOfShownSubpages/2);
-        }else{
-            startIndexToRender = 0;
-        }
+        addForwardsExtraNav(subpageNavigation)
     }
-    
+
+    nav.innerHTML = subpageNavigation.join("");
+
+    subpageNavAddOnClick(startIndexToRender, extraNavNeeded)
+
+    if(amountOfSubpages) {
+        document.querySelector(`.subpage-selector-index-${currentPageIndex}`).classList.add("selected");
+    }
+
+    updateSubpageInputBox();
+}
+
+function ShiftedStartIndexToRender(){
```

Comment:
Function naming: shouldn't have uppercase to begin with, also I don't understand what this name means.